python/ray/_raylet.pyx

```diff
@@ -2388,7 +2388,9 @@ cdef class GcsClient:
         object _nums_reconnect_retry
         CClusterID cluster_id
 
-    def __cinit__(self, address, nums_reconnect_retry=5, cluster_id=None):
+    def __cinit__(self, address,
+                  nums_reconnect_retry=RayConfig.instance().nums_gcs_reconnect_retry(),
```

Comment:
I feel like this may not work for workers (because gcs client is created before we initialize the ray config)?

Reply:
Let me print some numbers here so we'll know how things are going.